openassessment/data.py

```diff
@@ -670,3 +676,178 @@ def collect_ora2_responses(cls, course_id, desired_statuses=None):
                 result[item_id][status] += 1
 
         return result
+
+
+class OraDownloadData:
+    """
+    Helper class, that is used for downloading and compressing data related
+    to submissions (attachments, answer texts).
+    """
+
+    ATTACHMENT = 'attachment'
+    TEXT = 'text'
+
+    @classmethod
+    def _download_file_by_key(cls, key):
+        download_url = urljoin(
+            settings.LMS_ROOT_URL, get_download_url(key)
+        )
+
+        response = requests.get(download_url)
+        response.raise_for_status()
+
+        return response.content
+
+    @classmethod
+    def _create_downloads_csv(cls, submission_files_data):
+        """
+        Creates csv where each row represents all information about some certain
+        submission attachment or text.
+        """
+
+        header = (
+            'Course Id',
+            'Block Id',
+            'Student Id',
+            'File Key',
+            'File Name',
+            'File Type',
+            'File Description',
+            'File Size',
+            'Path',
+        )
+
+        rows = []
+
+        for file_data in submission_files_data:
+            rows.append(
+                (
+                    file_data['course_id'],
+                    file_data['block_id'],
+                    file_data['student_id'],
+                    file_data['key'],
+                    file_data['name'],
+                    file_data['type'],
+                    file_data['description'],
+                    file_data['size'],
+                    file_data['file_path'],
+                )
+            )
+
+        output_buffer = StringIO()
+        csvwriter = csv.writer(output_buffer)
+        csvwriter.writerows([header] + rows)
```

Comment:
Note: 

If you use DictWriter you can simplify this code a lot. This whole bit a lot. All this can become:
```suggestion
        output_buffer = StringIO()
        csvwriter = csv.DictWriter(output_buffer, raw_fields)
        csvwriter.writerow(header)
        csvwriter.writerows(file_data)
```

Reply:
I agree that [DictWriter](https://docs.python.org/3/library/csv.html#csv.DictWriter) would be cleaner. Since you're defining the column names in this class (`collect_ora2_submission_files`), you don't have to repeat them here.